pyogrio/raw.py

```diff
@@ -421,13 +418,13 @@ def write(
                 if not isinstance(v, str):
                     raise ValueError(f"metadata value {v} must be a string")
 
-    if promote_to_multi is None:
+    if geometry_type is not None and promote_to_multi is None:
```

Comment:
```suggestion
    if geometry is not None and promote_to_multi is None:
```

Reply:
I think it is a bit weird here to check on geometry because the code is about the geometry_type, but it won't influence anything so it doesn't matter so as you seem to have a preference I'll change it.